tests/nested/core20/kernel-failover/task.yaml

```diff
@@ -0,0 +1,89 @@
+summary: Check that a broken kernel snap automatically rolls itself back
+
+prepare: |
+  echo "Build a broken kernel snap where the initramfs panic's"
+
+  # shellcheck source=tests/lib/prepare.sh
+  . "$TESTSLIB/prepare.sh"
+
+  # shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+
+  # use the kernel snap from the nested VM and repack it so that it panic's in
+  # the initramfs and rolls back to the original version
+  # TODO:UC20: this doesn't work for some reason when we copy it from the 
+  #            running system, we should look into that, but for now just 
+  #            re-download the snap
+  snap download pc-kernel --channel=20/edge --basename=pc-kernel
+  rm -rf repacked-kernel
+  uc20_build_initramfs_kernel_snap pc-kernel.snap "$PWD" --inject-kernel-panic-in-initramfs
+  mv pc-kernel_*.snap panicking-initramfs-kernel.snap
+
+execute: |
+  # shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+
+  # Copy the broken initramfs kernel snap to the UC20 VM
+  copy_remote panicking-initramfs-kernel.snap
+
+  # wait for snapd to be "available"
```

Comment:
I think we will need a set new nested-utils tool set with thinks like this

Reply:
sure makes sense